src/v/coproc/script_context.cc

```diff
@@ -208,23 +203,26 @@ script_context::read_ntp(ss::lw_shared_ptr<ntp_context> ntp_ctx) {
           storage::log_reader_config cfg = get_reader(ntp_ctx);
           return ntp_ctx->log.make_reader(cfg)
             .then([](model::record_batch_reader rbr) {
-                return model::consume_reader_to_memory(
-                  std::move(rbr), model::no_timeout);
-            })
-            .then([](model::record_batch_reader::data_t data) {
-                return extract_batch_info(std::move(data));
+                return std::move(rbr).for_each_ref(
+                  coproc::reference_window_consumer(
+                    high_offset_tracker(),
+                    storage::internal::decompress_batch_consumer()),
+                  model::no_timeout);
             })
             .then(
-              [this, ntp_ctx](std::optional<batch_info> obatch_info)
```

Comment:
does it make sense to make these nested scopes independent functions ? seems a bit crammed w.r.t spacing.

Reply:
cleaning up

Reply:
very nice.